public/app/features/dashboard/dashgrid/DashboardLibrary/DashboardCard.tsx

```diff
@@ -0,0 +1,255 @@
+import { css, cx } from '@emotion/css';
+
+import { GrafanaTheme2 } from '@grafana/data';
+import { t, Trans } from '@grafana/i18n';
+import { Badge, Box, Button, Card, IconButton, Text, TextLink, Tooltip, useStyles2 } from '@grafana/ui';
+import { PluginDashboard } from 'app/types/plugins';
+
+import { GnetDashboard } from './types';
+
+interface Details {
+  id: string;
+  datasource: string;
+  dependencies: string[];
+  publishedBy: string;
+  lastUpdate: string;
+  grafanaComUrl?: string;
+}
+
+interface Props {
+  title: string;
+  imageUrl?: string;
+  dashboard: PluginDashboard | GnetDashboard;
+  details?: Details;
+  onClick: () => void;
+  isLogo?: boolean; // Indicates if imageUrl is a small logo vs full screenshot
+  showDatasourceProvidedBadge?: boolean;
+  dimThumbnail?: boolean; // Apply 50% opacity to thumbnail when badge is shown
+  showDescriptionPlaceholder?: boolean; // Show placeholder when no description exists
+  buttonText?: React.ReactNode; // Optional custom button text, defaults to "Use template"
+}
+
+export function DashboardCard({
+  title,
+  imageUrl,
+  onClick,
+  dashboard,
+  details,
+  isLogo,
+  showDatasourceProvidedBadge,
+  dimThumbnail,
+  showDescriptionPlaceholder,
+  buttonText,
+}: Props) {
+  const styles = useStyles2(getStyles);
+
+  return (
+    <Card className={styles.card} noMargin>
+      <Card.Heading className={styles.title}>{title}</Card.Heading>
+      <div className={isLogo ? styles.logoContainer : styles.thumbnailContainer}>
+        {imageUrl && imageUrl.trim() ? (
+          <img
+            src={imageUrl}
+            alt={title}
+            className={cx(
+              isLogo ? styles.logo : styles.thumbnail,
+              dimThumbnail && showDatasourceProvidedBadge && styles.dimmedImage
+            )}
+            onError={(e) => {
+              console.error('Failed to load image for:', title, 'URL:', imageUrl);
+              e.currentTarget.style.display = 'none';
+            }}
+          />
+        ) : (
+          <div className={styles.noImage}>
+            <Trans i18nKey="dashboard-template.card.no-preview">No preview available </Trans>
+          </div>
+        )}
+        {showDatasourceProvidedBadge && (
+          <div className={styles.badgeContainer}>
+            <Badge
+              text={t('dashboard-template.card.datasource-provided-badge', 'Data source provided')}
+              color="orange"
+            />
+          </div>
+        )}
+      </div>
+      <div title={dashboard.description || ''} className={styles.descriptionWrapper}>
+        {dashboard.description ? (
+          <Card.Description className={styles.description}>{dashboard.description}</Card.Description>
+        ) : showDescriptionPlaceholder ? (
+          <Card.Description className={cx(styles.description, styles.placeholderText)}>
+            <Trans i18nKey="dashboard-template.card.no-description">No description available</Trans>
+          </Card.Description>
```

Comment:
IMO we could probably nix this and just keep the same spacing as saying 'no description available' doesn't really add any value to end user 🤷‍♂️